components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/core/ParRequestBuilder.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.oauth.par.core;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.central.log.mgt.utils.LogConstants;
+import org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+import org.wso2.carbon.identity.oauth.par.common.ParConstants;
+import org.wso2.carbon.identity.oauth2.OAuthAuthorizationRequestBuilder;
+import org.wso2.carbon.utils.DiagnosticLog;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.wso2.carbon.identity.oauth.common.OAuthConstants.LogConstants.InputKeys.REQUEST_BUILDER;
+
+/**
+ * This builds the authorization request if the incoming request satisfies the PAR conditions.
+ */
+public class ParRequestBuilder implements OAuthAuthorizationRequestBuilder {
+
+    private static final Log log = LogFactory.getLog(ParRequestBuilder.class);
+
+    private static final String REQUEST_BUILDER_NAME = "Pushed authorization request builder";
+
+    @Override
+    public HttpServletRequest buildRequest(HttpServletRequest request) throws IdentityException {
+
+        return new OAuthParRequestWrapper(request);
+    }
+
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+
+        String requestUri = request.getParameter(OAuthConstants.OAuth20Params.REQUEST_URI);
+        boolean canHandle =
+                StringUtils.isNotBlank(requestUri) && requestUri.startsWith(ParConstants.REQUEST_URI_PREFIX);
```

Comment:
Can use StringUtils.startsWith it is null safe as well so we can omit the null check